### PR TITLE
Release notes 1.8.2 [6468]

### PR DESCRIPTION
**Pull request by LuisGP**

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/74/)

File: docs/notes.rst [hunk 1]
```diff
@@ -1,8 +1,42 @@
...
+* Fix for setting `nullptr` in a fixed string.
+* Fix for not sending GAP in several cases.
+* Solve *Coverity* report issues.
+* Fix issue of *fastrtpsgen* failing to open *IDL.g4* file.
+* Fix unnamed lock in *AESGCMGMAC_KeyFactory.cpp*.
+* Improve *XMLProfiles* example.
+* Multicast is now sent through *localhost* too.
+* *BitmapRange* now implements sliding window.
+* Improve *SequenceNumber_t* struct.
+* Participant's liveliness is now asserted when receive any RTPS message.
```
Comment (raquelalvarezbanos) [inline docs/notes.rst]:
```suggestion
* Participant's liveliness is now asserted when receiving any RTPS message.
```

File: docs/notes.rst [hunk 2]
```diff
@@ -1,8 +1,42 @@
...
+* Fix leak on RemoteParticipantLeaseDuration.
+* Modified default values to improve behavior in *Wi-Fi* scenarios.
+* *SubscriberHistory* improvements.
+* Removed use of *acceptMsgDirectTo*.
+* *WLP* improvements.
+
+**Note:** If you are upgrading from a version older than 1.7.0, it is **required** to regenerate generated source
+from IDL files using *fastrtpsgen*
```
Comment (raquelalvarezbanos) [inline docs/notes.rst]: I miss:

- Performance test improvements -> I assume this is intentional, as we said we wouldn't mention changes to tests
- Fix valgrind uninitialized error #743 -> not sure this is big enough to be mentioned
- Added copy constructor to ResourceLimitedVector -> not sure this is big enough to be mentioned

File: docs/notes.rst [hunk 3]
```diff
@@ -1,8 +1,42 @@
...
+It also adds the following bug fixes and improvements:
+
+* Fix for setting `nullptr` in a fixed string.
+* Fix for not sending GAP in several cases.
+* Solve *Coverity* report issues.
+* Fix issue of *fastrtpsgen* failing to open *IDL.g4* file.
+* Fix unnamed lock in *AESGCMGMAC_KeyFactory.cpp*.
+* Improve *XMLProfiles* example.
+* Multicast is now sent through *localhost* too.
+* *BitmapRange* now implements sliding window.
```
Comment (raquelalvarezbanos) [inline docs/notes.rst]: Can you confirm that the following:
- Fix issue of *fastrtpsgen* failing to open *IDL.g4* file
- Improve *XMLProfiles* example
- Multicast is now sent through localhost too
- BitmapRange now implements sliding window

are also part of 1.8.2, please?

File: docs/notes.rst [hunk 4]
```diff
@@ -1,8 +1,42 @@
...
+* Flag to avoid builtin multicast.
+* *STRICT_REALTIME* compilation flag.
+
+It also adds the following bug fixes and improvements:
+
+* Fix for setting `nullptr` in a fixed string.
+* Fix for not sending GAP in several cases.
+* Solve *Coverity* report issues.
+* Fix issue of *fastrtpsgen* failing to open *IDL.g4* file.
+* Fix unnamed lock in *AESGCMGMAC_KeyFactory.cpp*.
```
Comment (raquelalvarezbanos) [inline docs/notes.rst]: *AESGCMGMAC_KeyFactory.cpp*... What?? Was this the name of the method with the unnamed lock?

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/77/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/78/)

File: docs/notes.rst (hunk 2 above)
Comment (LuisGP) [reply docs/notes.rst]: > I miss:
> 
> * Performance test improvements -> I assume this is intentional, as we said we wouldn't mention changes to tests

Yes, it was intentional.

> * Fix valgrind uninitialized error #743 -> not sure this is big enough to be mentioned

This error was introduced between 1.8.1 and 1.8.2, so isn't a fix from the previous version, ti was a "development phase" error.

> * Added copy constructor to ResourceLimitedVector -> not sure this is big enough to be mentioned

`ResourceLimitedVector` is an internal class, not visible to the user. Am I wrong?

File: docs/notes.rst (hunk 3 above)
Comment (LuisGP) [reply docs/notes.rst]: > Can you confirm that the following:
> 
> * Fix issue of _fastrtpsgen_ failing to open _IDL.g4_ file
cfaf5307dae3ab94e6ff929f7507779585895691
> * Improve _XMLProfiles_ example
519b30ed4856c38943ee69c602224d3ea0a5e374
> * Multicast is now sent through localhost too
b45aaa41ca2427be25e0dc7f4e9756fc8a6fc37d
> * BitmapRange now implements sliding window
c447a38a2432ffeca3cda6d09e9022278f97f7e0
> 
> are also part of 1.8.2, please?

File: docs/notes.rst (hunk 2 above)
Comment (raquelalvarezbanos) [reply docs/notes.rst]: You're right. But do we only mention changes visible to the user? Maybe in this case it is not a big deal, so let's just not mention it.

File: docs/notes.rst (hunk 3 above)
Comment (raquelalvarezbanos) [reply docs/notes.rst]: Thanks!

File: docs/notes.rst (hunk 4 above)
Comment (LuisGP) [reply docs/notes.rst]: It is the name of the file.

**Comment by LuisGP** [comment]
Tested locally successfully.

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/249/)

**Pull request merged**